scipy/stats/stats.py

```diff
@@ -5718,3 +5720,179 @@ def rankdata(a, method='average'):
+    0.373298699196806752
+
+    If the bounding rectangle is not specified, it can be determined
+    numerically over the region specified by `bounds`.
+
+    >>> np.random.seed(12345)
+    >>> rvs = rvs_ratio_unif(f, size=2500, bounds=(-100, 100))
+    >>> stats.kstest(rvs, 'norm')[1]
+    0.3732986992033126
+
+    The exponential distribution provides another example where the bounding
+    rectangle can be determined explicitly.
+    >>> np.random.seed(12345)
+    >>> rvs = rvs_ratio_unif(lambda x: np.exp(-x), x1=0, x2=2*np.exp(-1), y2=1, size=1000)
+    >>> stats.kstest(rvs, 'expon')[1]
+    0.52369231518316373
+
+    Sometimes it can be helpful to use a non-zero shift parameter, see e.g.
+    [2]_ above in the case of the generalized inverse Gaussian distribution.
+
+    """
+
+    if not callable(f):
+        raise TypeError("pdf f must be callable.")
+
+    def find_rect_bounds(g, bounds):
+        if bounds is None:
+            raise Exception("bounds cannot be None if bounding rectangle is" +
+                            " not fully specified...")
+        opt_res = optimize.minimize_scalar(g, bounds=bounds, method='Bounded')
+        if not opt_res.success:
+            raise Exception("solver failed: " + opt_res.message)
+        if np.isnan(opt_res.fun):
+            raise Exception("solver returned nan at x=".format(opt_res.x))
+        return opt_res.fun
+
+    # in case bounding rectangle is not given, attempt to find it numerically
+    def f_opt(x):
+        return (x - shift) * np.sqrt(f(x))
+
+    if x1 is None:
+        x1 = find_rect_bounds(f_opt, bounds=bounds)
+    if x2 is None:
+        x2 = -find_rect_bounds(lambda x: -f_opt(x), bounds=bounds)
+
+    if x1 >= x2:
+        raise ValueError("x1 must be smaller than x2.")
+
+    if y2 is None:
+        y2 = np.sqrt(-find_rect_bounds(lambda x: -f(x), bounds=bounds))
+    if y2 < 0:
+        raise ValueError("y2 must be positive.")
+
+    if maxiter is None:
+        # set no of iterations to ensure that method fails with proba. p_fail
+        # in case point in [x1, x2] x [0, y2] is accepted with prob. p_accept
+        p_fail, p_accept = 0.001, 0.25
+        maxiter = 10 + size * np.log(1 / p_fail) / np.log(1 / (1 - p_accept))
+
+    # start sampling using ratio of uniforms method
+    x = np.zeros(size)
+    j, simulated = 1, 0
+    while (j <= maxiter):
+        k = size - simulated
+        # simulate uniform rvs on [x1, x2] and [0,y2]
+        u1 = distributions.uniform.rvs(loc=x1, scale=(x2 - x1), size=k)
+        v1 = distributions.uniform.rvs(scale=y2, size=k)
+        # apply rejection method
+        rvs = u1 / v1 + shift
+        accept = (v1**2 <= f(rvs))
+        num_accept = sum(accept)
+        if num_accept > 0:
+            take = min(num_accept, size - simulated)
+            x[simulated:(simulated + take)] = rvs[accept][0:take]
+            simulated += take
+        if simulated < size:
+            j += 1
+        else:
+            return x
```

Comment:
remember x needs to be N-Dimensional, depending on the `size` argument. I would use a vector for `x`, then reshape to the desired `size` when the function returns.